scripts/doc-generation/install-documentation-dependencies.sh

```diff
@@ -66,4 +67,18 @@ else
     step "Found Jazzy (${JAZZY_VERSION})"
 fi
 
+## Install xcodegen
```

Comment:
I almost feel like this entire script can be removed and we should just move it to generate-docs.sh

Reply:
having it separate was actually nice for testing it when I made this change

Reply:
I can see that use case, but why would this install script need to be reused? When would we specifically need to install these dependencies but not run the generate docs

Reply:
we don't have any other use cases for reusing the script at the moment. happy to combine it since it seems you feel strongly about it

Reply:
That would be great. I'd love to reduce unneeded overhead